From b7366b967a1c2ab26b5ae54b7ad43bfe0e3d1144 Mon Sep 17 00:00:00 2001
From: nE0sIghT <ykonotopov@gmail.com>
Date: Tue, 31 Mar 2015 19:45:18 +0300
Subject: [PATCH 3/4] Moved args check to proper function

---
 modules/wxwidgets.eselect | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/wxwidgets.eselect b/modules/wxwidgets.eselect
index c6468de..7b47dc8 100644
--- a/modules/wxwidgets.eselect
+++ b/modules/wxwidgets.eselect
@@ -26,8 +26,6 @@ describe_show() {
 }
 
 doshow_helper() {
-	[[ ${#@} -gt 0 ]] && die -q "Too many parameters"
-
 	write_list_start "Current ${multiarch}wxWidgets profile:"
 	if [[ -e ${WXCONFFILE} ]]; then
 		write_kv_list_entry "$(load_config ${WXCONFFILE} WXCONFIG${WXCONF_SUFFIX})" ""
@@ -37,6 +35,8 @@ doshow_helper() {
 }
 
 do_show() {
+	[[ ${#@} -gt 0 ]] && die -q "Too many parameters"
+
 	if [[ -d "${WXCONFDIR_32}" ]]; then
 		WXCONFDIR=${WXCONFDIR_64} multiarch="64bit " doshow_helper
 		WXCONFDIR=${WXCONFDIR_32} WXCONF_SUFFIX=_32 multiarch="32bit " doshow_helper
-- 
2.0.5

