From 6451ecb857d3ae9332c418693a8d4cf2b42e92a5 Mon Sep 17 00:00:00 2001
From: nE0sIghT <ykonotopov@gmail.com>
Date: Thu, 12 May 2016 22:28:04 +0300
Subject: [PATCH 2/3] Fixed corelib path

---
 gui/gui.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gui/gui.pro b/gui/gui.pro
index 869fed5..9adebd1 100644
--- a/gui/gui.pro
+++ b/gui/gui.pro
@@ -15,7 +15,7 @@ CONFIG += 64
 
 CONFIG += res_internal
 
-CORELIB_PATH = ../corelib
+CORELIB_PATH = ../corelib/corelib
 
 PRECOMPILED_HEADER = stdafx.h
 
-- 
2.7.3

