From 43379c075bc6b74359b63ba9df7e57febb7d787a Mon Sep 17 00:00:00 2001
From: nE0sIghT <ykonotopov@gmail.com>
Date: Thu, 12 May 2016 22:50:03 +0300
Subject: [PATCH 4/4] gui: added missing files

---
 gui/gui.pro | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gui/gui.pro b/gui/gui.pro
index 9adebd1..2cfe208 100644
--- a/gui/gui.pro
+++ b/gui/gui.pro
@@ -121,6 +121,7 @@ SOURCES += \
     utils/application.cpp \
     utils/gui_coll_helper.cpp \
     utils/InterConnector.cpp \
+    utils/IntValidator.cpp \
     utils/launch.cpp \
     utils/local_peer.cpp \
     utils/PainterPath.cpp \
@@ -281,6 +282,7 @@ HEADERS  += \
     utils/application.h \
     utils/gui_coll_helper.h \
     utils/InterConnector.h \
+    utils/IntValidator.cpp \
     utils/launch.h \
     utils/local_peer.h \
     utils/mac_support.h \
-- 
2.7.3

