From 0a0733800b8f9ddceea9f5f11d71380fb4ec2755 Mon Sep 17 00:00:00 2001
From: nE0sIghT <ykonotopov@gmail.com>
Date: Tue, 31 Mar 2015 19:31:31 +0300
Subject: [PATCH 2/4] Removed debug typo

---
 modules/wxwidgets.eselect | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/wxwidgets.eselect b/modules/wxwidgets.eselect
index 400b5f9..c6468de 100644
--- a/modules/wxwidgets.eselect
+++ b/modules/wxwidgets.eselect
@@ -140,7 +140,7 @@ doset_helper() {
 
 do_set() {
 	[[ ${#@} -eq 0 ]] \
-		&& die -q "You must specify a profile name or number: ${#@}"
+		&& die -q "You must specify a profile name or number"
 
 	[[ ! -w "${EROOT}"/var/lib/ ]] \
 		&& die -q "You need write permission to /var/lib to perform this action."
-- 
2.0.5

