diff -ur ./CorsixTH-0.30/CorsixTH/CMakeLists.txt ./CorsixTH-0.30-nodoc/CorsixTH/CMakeLists.txt
--- ./CorsixTH-0.30/CorsixTH/CMakeLists.txt	2013-12-01 20:49:04.000000000 +0400
+++ ./CorsixTH-0.30-nodoc/CorsixTH/CMakeLists.txt	2014-05-06 19:20:22.509724777 +0400
@@ -228,5 +228,5 @@
   install(DIRECTORY Bitmap DESTINATION CorsixTH
         FILES_MATCHING REGEX ".*\\.(tab|pal|dat|png)$"
         PATTERN "*.svn" EXCLUDE)
-  install(FILES CorsixTH.lua LICENSE.txt CorsixTH.ico DESTINATION CorsixTH )
+  install(FILES CorsixTH.lua CorsixTH.ico DESTINATION CorsixTH )
 ENDIF()
