diff -ur ./s25rttr-9999/libutil/src/files.h ./s25rttr-9999.1/libutil/src/files.h
--- ./s25rttr-9999/libutil/src/files.h	2014-04-28 21:27:20.984616000 +0400
+++ ./s25rttr-9999.1/libutil/src/files.h	2014-04-28 21:28:40.453783225 +0400
@@ -140,7 +140,7 @@
     /* 54 */RTTRDIR "",                               // unbenutzt
     /* 55 */SETTINGSDIR "/LSTS/SOUND.LST",            // Die konvertierte sound.lst
     /* 56 */RTTRDIR "/sound.scs",                     // Das konvertier-script
-    /* 57 */RTTRDIR "",                               // Basispfad fürn Soundconverter
+    /* 57 */DRIVERDIR "",                             // Basispfad fürn Soundconverter
     /* 58 */GAMEDIR "/DATA/MIS0BOBS.LST",
     /* 59 */GAMEDIR "/DATA/MIS1BOBS.LST",
     /* 60 */GAMEDIR "/DATA/MIS2BOBS.LST",
