From 975950374879599829cc23ff0131aeb7c0886a17 Mon Sep 17 00:00:00 2001
From: nE0sIghT <ykonotopov@gmail.com>
Date: Tue, 22 Sep 2015 10:44:55 +0300
Subject: [PATCH 3/3] Use shared boost libraries

---
 cmake/linux.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/linux.cmake b/cmake/linux.cmake
index 9ab448c..d4d8549 100644
--- a/cmake/linux.cmake
+++ b/cmake/linux.cmake
@@ -1,4 +1,4 @@
-SET(Boost_USE_STATIC_LIBS TRUE)
+SET(Boost_USE_STATIC_LIBS FALSE)
 
 #Check for ccache
 find_program(CCACHE_FOUND ccache)
-- 
2.5.3

