diff -ur ./pcsx2-1.2.2/cmake/BuildParameters.cmake ./pcsx2-1.2.2-r1/cmake/BuildParameters.cmake
--- ./pcsx2-1.2.2/cmake/BuildParameters.cmake	2014-02-10 22:57:09.000000000 +0400
+++ ./pcsx2-1.2.2-r1/cmake/BuildParameters.cmake	2014-09-27 21:55:02.178118323 +0400
@@ -151,6 +151,10 @@
         set(GAMEINDEX_DIR "${CMAKE_INSTALL_PREFIX}/share/games/pcsx2")
     endif(NOT DEFINED GAMEINDEX_DIR)
 
+    if(NOT DEFINED BIN_DIR)
+        set(BIN_DIR "${CMAKE_INSTALL_PREFIX}/bin")
+    endif(NOT DEFINED BIN_DIR)
+
     # Compile all source codes with these 3 defines
     add_definitions(-DPLUGIN_DIR_COMPILATION=${PLUGIN_DIR} -DGAMEINDEX_DIR_COMPILATION=${GAMEINDEX_DIR})
 endif(PACKAGE_MODE)
diff -ur ./pcsx2-1.2.2/pcsx2/CMakeLists.txt ./pcsx2-1.2.2-r1/pcsx2/CMakeLists.txt
--- ./pcsx2-1.2.2/pcsx2/CMakeLists.txt	2014-02-10 22:57:09.000000000 +0400
+++ ./pcsx2-1.2.2-r1/pcsx2/CMakeLists.txt	2014-09-27 21:55:28.298690560 +0400
@@ -688,7 +688,7 @@
 endif(NOT USER_CMAKE_LD_FLAGS STREQUAL "")
 
 if(PACKAGE_MODE)
-    install(TARGETS ${Output} DESTINATION bin)
+    install(TARGETS ${Output} DESTINATION ${BIN_DIR})
 else(PACKAGE_MODE)
     install(TARGETS ${Output} DESTINATION ${CMAKE_SOURCE_DIR}/bin)
 endif(PACKAGE_MODE)
diff -ur ./pcsx2-1.2.2/plugins/GSdx/CMakeLists.txt ./pcsx2-1.2.2-r1/plugins/GSdx/CMakeLists.txt
--- ./pcsx2-1.2.2/plugins/GSdx/CMakeLists.txt	2014-02-10 22:57:09.000000000 +0400
+++ ./pcsx2-1.2.2-r1/plugins/GSdx/CMakeLists.txt	2014-09-27 21:55:35.185508099 +0400
@@ -230,7 +230,7 @@
 	endif(NOT USER_CMAKE_LD_FLAGS STREQUAL "")
 
 	if(PACKAGE_MODE)
-	    install(TARGETS ${Replay} DESTINATION bin)
+	    install(TARGETS ${Replay} DESTINATION ${BIN_DIR})
 	else(PACKAGE_MODE)
 	    install(TARGETS ${Replay} DESTINATION ${CMAKE_SOURCE_DIR}/bin)
 	endif(PACKAGE_MODE)
diff -ur ./pcsx2-1.2.2/plugins/zzogl-pg/opengl/CMakeLists.txt ./pcsx2-1.2.2-r1/plugins/zzogl-pg/opengl/CMakeLists.txt
--- ./pcsx2-1.2.2/plugins/zzogl-pg/opengl/CMakeLists.txt	2014-02-10 22:57:09.000000000 +0400
+++ ./pcsx2-1.2.2-r1/plugins/zzogl-pg/opengl/CMakeLists.txt	2014-09-27 21:55:48.479132663 +0400
@@ -257,7 +257,7 @@
 	endif(NOT USER_CMAKE_LD_FLAGS STREQUAL "")
 
 	if(PACKAGE_MODE)
-	    install(TARGETS ${Replay} DESTINATION bin)
+	    install(TARGETS ${Replay} DESTINATION ${BIN_DIR})
 	else(PACKAGE_MODE)
 	    install(TARGETS ${Replay} DESTINATION ${CMAKE_SOURCE_DIR}/bin)
 	endif(PACKAGE_MODE)
diff -ur ./pcsx2-1.2.2/plugins/zzogl-pg-cg/opengl/CMakeLists.txt ./pcsx2-1.2.2-r1/plugins/zzogl-pg-cg/opengl/CMakeLists.txt
--- ./pcsx2-1.2.2/plugins/zzogl-pg-cg/opengl/CMakeLists.txt	2014-02-10 22:57:09.000000000 +0400
+++ ./pcsx2-1.2.2-r1/plugins/zzogl-pg-cg/opengl/CMakeLists.txt	2014-09-27 21:55:41.558981061 +0400
@@ -220,7 +220,7 @@
 	endif(NOT USER_CMAKE_LD_FLAGS STREQUAL "")
 
 	if(PACKAGE_MODE)
-	    install(TARGETS ${Replay} DESTINATION bin)
+	    install(TARGETS ${Replay} DESTINATION ${BIN_DIR})
 	else(PACKAGE_MODE)
 	    install(TARGETS ${Replay} DESTINATION ${CMAKE_SOURCE_DIR}/bin)
 	endif(PACKAGE_MODE)
