diff -ur pcsx2-1.2.2/cmake/FindCg.cmake pcsx2-1.2.2-r1/cmake/FindCg.cmake
--- pcsx2-1.2.2/cmake/FindCg.cmake	2014-02-10 22:57:09.000000000 +0400
+++ pcsx2-1.2.2-r1/cmake/FindCg.cmake	2014-05-18 19:25:34.100162594 +0400
@@ -52,6 +52,7 @@
 				 /usr/include
 				 /usr/local/lib
 				 /opt/nvidia-cg-toolkit/lib	
+				 /opt/nvidia-cg-toolkit/lib32
 				 DOC "Path to the Cg library.")
 
 	# Cg GL library
@@ -59,6 +60,7 @@
 				 /usr/include
 				 /usr/local/lib
 				 /opt/nvidia-cg-toolkit/lib
+				 /opt/nvidia-cg-toolkit/lib32
 				 DOC "Path to the CgGL library.")
 
 	set(CG_LIBRARIES ${CG_LIBRARY} ${CG_GL_LIBRARY})
