diff -ur pcsc-lite-1.8.13/etc/pcscd.service.in pcsc-lite-1.8.13-r1/etc/pcscd.service.in
--- pcsc-lite-1.8.13/etc/pcscd.service.in	2011-11-26 01:36:24.000000000 +0400
+++ pcsc-lite-1.8.13-r1/etc/pcscd.service.in	2015-04-03 19:03:53.000000000 +0300
@@ -5,6 +5,8 @@
 [Service]
 ExecStart=@sbindir_exp@/pcscd --foreground --auto-exit
 ExecReload=@sbindir_exp@/pcscd --hotplug
+User=pcscd
+Group=pcscd
 
 [Install]
 Also=pcscd.socket
